File: Arrays and Hashing/two-sum.py
# time O(n) space O(n)
class Solution:
    def twoSum(self, nums, target):
        hashMap = {}
        for i, num in enumerate(nums):
            diff = target - num
            if diff in hashMap:
                return [i, hashMap[diff]]
            hashMap[num] = i

File: Arrays and Hashing/two-sum.js
// time O(n) space O(n)
var twoSum = function (nums, target) {
  const lookUp = {};
  for (let i = 0; i < nums.length; i++) {
    let diff = target - nums[i];
    if (lookUp[diff] !== undefined) {
      return [i, lookUp[diff]];
    } else {
      lookUp[nums[i]] = i;
    }
  }
};
